pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java

```diff
@@ -142,6 +145,8 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
         // Single-value
         if (metadata.isSorted()) {
           // Sorted
+          // forwardIndexDisabled columns do not need to be handled here as forward index cannot be disabled on a
```

Comment:
May be I am mis-reading the if-else block here. For an unsorted dict column with inv index, are we ok with line 157 being executed with null `fwdIndexBuffer` ?

Reply:
yes it is okay for 157 to be called with `fwdIndexBuffer` as null. We only set this to null if the forward index is disabled. When the forward index reader is created, the No-op forward index reader will be assigned which doesn't need a `fwdIndexBuffer`. I've added a comment where I set `fwdIndexBuffer` to indicate this behavior is expected. Let me know if you would like more clarification on this.